src/matvis/core/matprod.py

```diff
@@ -29,15 +32,24 @@ def __init__(
         nchunks: int,
         nfeed: int,
         nant: int,
-        antpairs: np.ndarray | None,
+        antpairs: np.ndarray | list | None,
+        matsets: list | None,
         precision=1,
     ):
         if antpairs is None:
             self.all_pairs = True
             self.antpairs = np.array([(i, j) for i in range(nant) for j in range(nant)])
+            self.matsets = list(
+                [
+                    (np.array([i]), np.array([j]))
+                    for i in range(nant)
+                    for j in range(nant)
+                ]
+            )
```

Comment:
I think maybe by default the matsets can be left as None in the base class?

Reply:
yeah, agreed.  done